src/System.Collections.NonGeneric/tests/Performance/Perf.HashTable.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+using Microsoft.Xunit.Performance;
+using System.Collections.Generic;
+
+namespace System.Collections.Tests
+{
+    public class Perf_HashTable
+    {
+        public static IEnumerable<object[]> TestHashtables()
+        {
+            yield return new object[] { CreateHashtable(100) };
+            yield return new object[] { CreateHashtable(1000) };
+        }
+
+        public static Hashtable CreateHashtable(int size)
+        {
+            Hashtable ht = new Hashtable();
+            for (int i = 0; i < size; i++)
+                ht.Add(PerfUtils.CreateString(50), PerfUtils.CreateString(50));
+            return ht;
+        }
+
+        [Benchmark]
+        public void ctor()
+        {
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                {
+                    new Hashtable(); new Hashtable(); new Hashtable(); new Hashtable(); new Hashtable();
+                    new Hashtable(); new Hashtable(); new Hashtable(); new Hashtable(); new Hashtable();
+                    new Hashtable(); new Hashtable(); new Hashtable(); new Hashtable(); new Hashtable();
+                    new Hashtable(); new Hashtable(); new Hashtable(); new Hashtable(); new Hashtable();
+                }
+        }
+
+        [Benchmark]
+        [MemberData("TestHashtables")]
+        public void GetItem(Hashtable table)
+        {
+            object result;
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                table.Add("key", "value");
```

Comment:
>and then get rid of the Remove?

This I'm a bit torn on. In their current state the tests will not modify the collections passed to them as MemberData. I wrote them this way so that we could potentially cache the collections used as MemberData to greatly speed up per-test initialization. However, I'm a bit concerned about future-proofing and possible errors: if a test is added that modifies a MemberData collection that is later used by a different test, it could skew those results.

What do you think? Is the speed-up worth the risk for error?